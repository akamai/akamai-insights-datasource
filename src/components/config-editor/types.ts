export enum Secret {
  ClientSecret = 'clientSecret',
  Host = 'host',
  AccessToken = 'accessToken',
  ClientToken = 'clientToken'
}
