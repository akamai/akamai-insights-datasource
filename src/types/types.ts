import { DataSourceJsonData, DataSourcePluginOptionsEditorProps } from '@grafana/data';
import { DataQuery } from '@grafana/schema';

export interface MyQuery extends DataQuery {
  reportLink?: string;
  dimensions?: string[];
  metrics?: string[];
  filters?: FilterQuery[];
  sortBys?: SortByQuery[];
  limit?: number;
}

export interface FilterQuery {
  name?: string;
  operator?: string;
  expressions?: ExpressionType[];
  expression?: ExpressionType;
}

export interface SortByQuery {
  name?: string;
  sortOrder?: string;
}

export type ExpressionType = string | number;

export interface MyDataSourceOptions extends DataSourceJsonData {
  clientSecret?: string;
  host?: string;
  accessToken?: string;
  clientToken?: string;
}

export interface DataSourceProps extends DataSourcePluginOptionsEditorProps<MyDataSourceOptions> {}

export enum TestDataSourceResponseStatus {
  Success = 'success',
  Error = 'error'
}

export interface NameAware {
  name: string
}

export interface Enum {
  [id: string]: string;
}

export const DATA_REQUEST_LIMIT = 50000;
