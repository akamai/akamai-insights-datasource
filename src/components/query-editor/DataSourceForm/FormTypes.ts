import { SelectableValue } from '@grafana/data';
import { groupBy } from 'lodash';

import { enumToSelectableValues, prettyEnum, stringsToSelectableValues } from '../../../utils/utils';

export const shortLabelWidth = 25;
export const mediumLabelWidth = 50;

export enum FilterType {
  Dimension = 'Dimension',
  Metric = 'Metric'
}

export const filterTypeOptions = enumToSelectableValues(FilterType);

export const filterTypeOptionsMap = groupBy(filterTypeOptions, 'value');

export interface FilterFormModel {
  type: SelectableValue<string>;
  query: FilterQueryFormModel;
}

export interface FilterQueryFormModel {
  name?: SelectableValue<string>;
  operator?: SelectableValue<string>;
  expressions?: string[];
}

export enum DimensionOperator {
  InList = 'IN_LIST',
  RegExp = 'REGEXP',
  BeginsWith = 'BEGINS_WITH',
  EndsWith = 'ENDS_WITH',
  Partial = 'PARTIAL',
  Exact = 'EXACT'
}

export enum FilterValueType {
  Text = 'TEXT',
  Enum = 'ENUM'
}

export const dimensionOperatorOptions = enumToSelectableValues(DimensionOperator);

export const authorizableOrEnumFiltersOperatorOptions = stringsToSelectableValues([ DimensionOperator.InList ], prettyEnum);

export enum MetricOperator {
  Equal = 'EQUAL',
  NotEqual = 'NOT_EQUAL',
  LessThan = 'LESS_THAN',
  GreaterThan = 'GREATER_THAN'
}

export const metricOperatorOptions = enumToSelectableValues(MetricOperator);

export interface SortByFormModel {
  type?: SelectableValue<string>;
  query: SortByQueryFormModel;
}

export interface SortByQueryFormModel {
  name?: SelectableValue<string>;
  sortOrder?: SelectableValue<string>;
}

export enum SortByOrder {
  Ascending = 'ASCENDING',
  Descending = 'DESCENDING'
}

export const sortByOrderOptions = enumToSelectableValues(SortByOrder);
