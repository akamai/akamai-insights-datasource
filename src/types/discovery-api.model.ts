import { DataQuery } from '@grafana/schema';

export interface DiscoveryApiModel extends DataQuery {
  metrics: Metric[];
  dimensions: Dimension[];
  defaults: Defaults[];
}

export interface Metric {
  name: string;
  type: string;
}

export interface Dimension {
  name: string;
  type: string;
  filterable: boolean;
  filterType: string;
  authorizable: boolean;
  filterEnumValues?: string[];
}

export interface Defaults {
  defaultTimeRange: DefaultTimeRange;
  defaultMetrics: string[];
  defaultDimensions: string[];
  defaultSortBys: DefaultSortBy[];
}

export interface DefaultTimeRange {
  start: string;
  end: string;
}

export interface DefaultSortBy {
  name: string;
  sortOrder: 'ASCENDING' | 'DESCENDING';
}

export const initialModel: DiscoveryApiModel = {
  metrics: [],
  dimensions: [],
  defaults: [],
  refId: ''
};
