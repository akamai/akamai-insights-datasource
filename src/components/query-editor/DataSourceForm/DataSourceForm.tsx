import { SelectableValue } from '@grafana/data';
import { Button, InlineField, InlineFieldRow, MultiSelect } from '@grafana/ui';
import React, { JSX, useState } from 'react';

import { DatasourceService } from '../../../services/datasource.service';
import { FormService } from '../../../services/form.service';
import { DiscoveryApiModel } from '../../../types/discovery-api.model';
import { MyQuery } from '../../../types/types';
import { stringsToSelectableValues, toSelectableValues } from '../../../utils/utils';
import { FilterEditorRow } from './FilterEditor/FilterEditorRow';
import { FilterFormModel, mediumLabelWidth, shortLabelWidth, SortByFormModel } from './FormTypes';
import { ListField } from './ListField/ListField';
import { SortByEditorRow } from './SortEditor/SortByEditorRow';

interface DataSourceFormProps {
  model: DiscoveryApiModel,
  datasource: DatasourceService,
  query: MyQuery,
  onRunQuery: () => void,
  onChange: (value: MyQuery) => void
}

export function DataSourceForm({ query, onChange, onRunQuery, model }: DataSourceFormProps): JSX.Element {
  const dimensionsOptions = toSelectableValues(model.dimensions);
  const metricsOptions = toSelectableValues(model.metrics);
  const dimensionsValues = FormService.toValues(dimensionsOptions);
  const metricsValues = FormService.toValues(metricsOptions);

  const [ dimensions, setDimensions ] = useState<SelectableValue[]>(model.dimensions ? stringsToSelectableValues(query.dimensions) : []);
  const [ metrics, setMetrics ] = useState<SelectableValue[]>(model.metrics ? stringsToSelectableValues(query.metrics) : []);
  const [ filters, setFilters ] = useState<FilterFormModel[]>(model.metrics && model.dimensions ? FormService.toFilterFormModels(query.filters, dimensionsValues, metricsValues) : []);
  const [ sortBys, setSortBys ] = useState<SortByFormModel[]>(model.metrics && model.dimensions ? FormService.toSortBysFormModel(query.sortBys, dimensionsValues, metricsValues) : []);

  const onClear = () => {
    setDimensions([]);
    setMetrics([]);
    setFilters([]);
    setSortBys([]);
  };

  const onApply = () => {
    const updatedQuery = {
      ...query
    };

    updatedQuery.dimensions = FormService.toValues(dimensions);
    updatedQuery.metrics = FormService.toValues(metrics);
    updatedQuery.filters = FormService.toFilterQueries(filters);
    updatedQuery.sortBys = FormService.toSortByQueries(sortBys);

    onChange(updatedQuery);
    onRunQuery();
  };

  return (
    <div>
      <InlineField
        label="Dimensions"
        labelWidth={shortLabelWidth}
        tooltip="Array of dimensions for grouping results (i.e.['time5minutes', 'cpcode']). Maximum number of dimensions: 4.">
        <MultiSelect
          isClearable={true}
          width={mediumLabelWidth}
          options={dimensionsOptions}
          value={dimensions}
          onChange={setDimensions}
        />
      </InlineField>

      <InlineField
        label="Metrics"
        labelWidth={shortLabelWidth}
        tooltip="Array of metrics requested (i.e.: ['edgeBytesSum', 'originHitsSum']).">
        <MultiSelect
          isClearable={true}
          width={mediumLabelWidth}
          options={metricsOptions}
          value={metrics}
          onChange={setMetrics}
        />
      </InlineField>

      <ListField
        fieldLabel="Filters"
        fieldTooltip="Filters used to narrow down the results of the data. Filters are specified in terms of dimensions or metrics."
        resource="Filter"
        rows={filters}
        onChange={setFilters}
        modelProvider={() => FormService.creteEmptyFilter(dimensionsOptions[ 0 ])}
        Editor={(filterModel, index, onSingleFilterChange) =>
          <FilterEditorRow
            key={index}
            model={filterModel}
            dimensions={dimensionsOptions}
            metrics={metricsOptions}
            onChange={updatedFilterModel => onSingleFilterChange(updatedFilterModel, index)}
          />}
      />

      <ListField
        fieldLabel="Sort Bys"
        fieldTooltip="Set sortBy settings for multiple columns. If not specified, the default values are used."
        resource="Sort By"
        rows={sortBys}
        onChange={setSortBys}
        modelProvider={() => FormService.creteEmptySortBy(dimensionsOptions[ 0 ])}
        Editor={(sortByModel, index, onSingleSortByChange) =>
          <SortByEditorRow
            key={index}
            model={sortByModel}
            dimensions={dimensionsOptions}
            metrics={metricsOptions}
            onChange={updatedSortByModel => onSingleSortByChange(updatedSortByModel, index)}
          />}
      />

      <InlineFieldRow>
        <InlineField>
          <Button
            fill="outline"
            type="reset"
            variant="destructive"
            onClick={onClear}>
            Clear
          </Button>
        </InlineField>
        <InlineField>
          <Button
            type="submit"
            variant="primary"
            onClick={onApply}>
            Apply
          </Button>
        </InlineField>
      </InlineFieldRow>
    </div>
  );
}
