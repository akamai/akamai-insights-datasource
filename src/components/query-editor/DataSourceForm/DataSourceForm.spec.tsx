import React from 'react';
import renderer from 'react-test-renderer';

import { Dimension, DiscoveryApiModel, Metric } from '../../../types/discovery-api.model';
import { MyQuery } from '../../../types/types';
import { DataSourceForm } from './DataSourceForm';

const datasource: any = {};
const model: DiscoveryApiModel = {
  dimensions: [
    {
      name: 'cpcode'
    } as Dimension
  ],
  metrics: [
    {
      name: 'edgeHitsSum'
    } as Metric
  ],
  datasource
} as any as DiscoveryApiModel;

describe('DataSourceForm component', () => {
  it('should render new empty editor', () => {
    const query = {
      dimensions: [],
      metrics: [],
      datasource
    } as any as MyQuery;

    const onChange = jest.fn();
    const onRunQuery = jest.fn();

    const tree = renderer.create(
      <DataSourceForm datasource={datasource} query={query} model={model} onChange={onChange} onRunQuery={onRunQuery}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render existing edit editor', () => {
    const query = {
      dimensions: [ 'cpcode' ],
      metrics: [ 'edgeHitsSum' ],
      datasource
    } as any as MyQuery;

    const onChange = jest.fn();
    const onRunQuery = jest.fn();

    const tree = renderer.create(
      <DataSourceForm datasource={datasource} query={query} model={model} onChange={onChange} onRunQuery={onRunQuery}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
