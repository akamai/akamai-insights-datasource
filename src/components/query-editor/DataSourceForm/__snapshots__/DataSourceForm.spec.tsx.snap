// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSourceForm component should render existing edit editor 1`] = `
<div>
  <div
    className="css-1f5gueu"
  >
    <label
      className="css-1e5y6i9"
    >
      Dimensions
      <div
        className="css-wf08df-Icon"
        tabIndex={0}
      >
        <svg
          data-testid="info-circle"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        />
      </div>
    </label>
    <div
      className="css-1rfunm5"
    >
      <div
        className="css-jjhwg3-input-wrapper css-17r2o28-SelectContainer"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-4-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className="css-1scgfe8"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="css-1eq04kg-grafana-select-value-container-grafana-select-value-container-multi"
          >
            <div
              className="css-1idmvrx-grafana-select-multi-value-container"
              css={{}}
            >
              <div
                className=" css-1l9kcyg-MultiValueGeneric"
              >
                cpcode
              </div>
              <button
                aria-label="Remove cpcode"
                className="css-1l39x10-grafana-select-multi-value-remove"
                css={{}}
                onClick={[Function]}
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <div
                  className="css-wf08df-Icon"
                >
                  <svg
                    data-testid="times"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  />
                </div>
              </button>
            </div>
            <div
              className=" css-lq7btu-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-expanded={false}
                aria-haspopup={true}
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-4-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="css-uvldi-input-suffix"
          >
            <div
              aria-label="select-clear-value"
              className="css-wf08df-Icon"
              onMouseDown={[Function]}
              role="button"
            >
              <svg
                data-testid="times"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              />
            </div>
            <div
              className="css-wf08df-Icon"
            >
              <svg
                data-testid="angle-down"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-1f5gueu"
  >
    <label
      className="css-1e5y6i9"
    >
      Metrics
      <div
        className="css-wf08df-Icon"
        tabIndex={0}
      >
        <svg
          data-testid="info-circle"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        />
      </div>
    </label>
    <div
      className="css-1rfunm5"
    >
      <div
        className="css-jjhwg3-input-wrapper css-17r2o28-SelectContainer"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-5-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className="css-1scgfe8"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="css-1eq04kg-grafana-select-value-container-grafana-select-value-container-multi"
          >
            <div
              className="css-1idmvrx-grafana-select-multi-value-container"
              css={{}}
            >
              <div
                className=" css-1l9kcyg-MultiValueGeneric"
              >
                edgeHitsSum
              </div>
              <button
                aria-label="Remove edgeHitsSum"
                className="css-1l39x10-grafana-select-multi-value-remove"
                css={{}}
                onClick={[Function]}
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <div
                  className="css-wf08df-Icon"
                >
                  <svg
                    data-testid="times"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  />
                </div>
              </button>
            </div>
            <div
              className=" css-lq7btu-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-expanded={false}
                aria-haspopup={true}
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-5-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="css-uvldi-input-suffix"
          >
            <div
              aria-label="select-clear-value"
              className="css-wf08df-Icon"
              onMouseDown={[Function]}
              role="button"
            >
              <svg
                data-testid="times"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              />
            </div>
            <div
              className="css-wf08df-Icon"
            >
              <svg
                data-testid="angle-down"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-9doeo5-InlineFieldRow"
  >
    <div
      className="css-1f5gueu"
    >
      <label
        className="css-1e5y6i9"
      >
        Filters
        <div
          className="css-wf08df-Icon"
          tabIndex={0}
        >
          <svg
            data-testid="info-circle"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          />
        </div>
      </label>
      <div
        className="css-1rfunm5"
      >
        <div
          className="css-1uedr6a"
        >
          <button
            aria-label="Add new Filter"
            className="css-x1vujn"
            onClick={[Function]}
            tabIndex={0}
          >
            <div
              className="css-wf08df-Icon"
            >
              <svg
                data-testid="plus-circle"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              />
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-9doeo5-InlineFieldRow"
  >
    <div
      className="css-1f5gueu"
    >
      <label
        className="css-1e5y6i9"
      >
        Sort Bys
        <div
          className="css-wf08df-Icon"
          tabIndex={0}
        >
          <svg
            data-testid="info-circle"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          />
        </div>
      </label>
      <div
        className="css-1rfunm5"
      >
        <div
          className="css-1uedr6a"
        >
          <button
            aria-label="Add new Sort By"
            className="css-x1vujn"
            onClick={[Function]}
            tabIndex={0}
          >
            <div
              className="css-wf08df-Icon"
            >
              <svg
                data-testid="plus-circle"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              />
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-1f5gueu"
  >
    <label
      className="css-1e5y6i9"
    >
      Limit
      <div
        className="css-wf08df-Icon"
        tabIndex={0}
      >
        <svg
          data-testid="info-circle"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        />
      </div>
    </label>
    <div
      className="css-1rfunm5"
    >
      <div
        className="css-ax7m4m-input-wrapper"
        data-testid="input-wrapper"
      >
        <div
          className="css-1w5c5dq-input-inputWrapper"
        >
          <input
            className="css-1n6y0bv-input-input"
            max={50000}
            min={0}
            onChange={[Function]}
            style={
              {
                "paddingLeft": undefined,
                "paddingRight": undefined,
              }
            }
            type="number"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-9doeo5-InlineFieldRow"
  >
    <div
      className="css-1f5gueu"
    >
      <div
        className="css-1rfunm5"
      >
        <button
          className="css-1oe1lwi-button"
          onClick={[Function]}
          type="reset"
        >
          <span
            className="css-1mhnkuh"
          >
            Clear
          </span>
        </button>
      </div>
    </div>
    <div
      className="css-1f5gueu"
    >
      <div
        className="css-1rfunm5"
      >
        <button
          className="css-td06pi-button"
          onClick={[Function]}
          type="submit"
        >
          <span
            className="css-1mhnkuh"
          >
            Apply
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`DataSourceForm component should render new empty editor 1`] = `
<div>
  <div
    className="css-1f5gueu"
  >
    <label
      className="css-1e5y6i9"
    >
      Dimensions
      <div
        className="css-wf08df-Icon"
        tabIndex={0}
      >
        <svg
          data-testid="info-circle"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        />
      </div>
    </label>
    <div
      className="css-1rfunm5"
    >
      <div
        className="css-jjhwg3-input-wrapper css-17r2o28-SelectContainer"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-2-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className="css-1scgfe8"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="css-1eq04kg-grafana-select-value-container-grafana-select-value-container-multi"
          >
            <div
              className=" css-1n8tjau-placeholder"
              id="react-select-2-placeholder"
            >
              Choose
            </div>
            <div
              className=" css-lq7btu-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-describedby="react-select-2-placeholder"
                aria-expanded={false}
                aria-haspopup={true}
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-2-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="css-uvldi-input-suffix"
          >
            <div
              className="css-wf08df-Icon"
            >
              <svg
                data-testid="angle-down"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-1f5gueu"
  >
    <label
      className="css-1e5y6i9"
    >
      Metrics
      <div
        className="css-wf08df-Icon"
        tabIndex={0}
      >
        <svg
          data-testid="info-circle"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        />
      </div>
    </label>
    <div
      className="css-1rfunm5"
    >
      <div
        className="css-jjhwg3-input-wrapper css-17r2o28-SelectContainer"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-3-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className="css-1scgfe8"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="css-1eq04kg-grafana-select-value-container-grafana-select-value-container-multi"
          >
            <div
              className=" css-1n8tjau-placeholder"
              id="react-select-3-placeholder"
            >
              Choose
            </div>
            <div
              className=" css-lq7btu-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-describedby="react-select-3-placeholder"
                aria-expanded={false}
                aria-haspopup={true}
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-3-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="css-uvldi-input-suffix"
          >
            <div
              className="css-wf08df-Icon"
            >
              <svg
                data-testid="angle-down"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-9doeo5-InlineFieldRow"
  >
    <div
      className="css-1f5gueu"
    >
      <label
        className="css-1e5y6i9"
      >
        Filters
        <div
          className="css-wf08df-Icon"
          tabIndex={0}
        >
          <svg
            data-testid="info-circle"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          />
        </div>
      </label>
      <div
        className="css-1rfunm5"
      >
        <div
          className="css-1uedr6a"
        >
          <button
            aria-label="Add new Filter"
            className="css-x1vujn"
            onClick={[Function]}
            tabIndex={0}
          >
            <div
              className="css-wf08df-Icon"
            >
              <svg
                data-testid="plus-circle"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              />
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-9doeo5-InlineFieldRow"
  >
    <div
      className="css-1f5gueu"
    >
      <label
        className="css-1e5y6i9"
      >
        Sort Bys
        <div
          className="css-wf08df-Icon"
          tabIndex={0}
        >
          <svg
            data-testid="info-circle"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          />
        </div>
      </label>
      <div
        className="css-1rfunm5"
      >
        <div
          className="css-1uedr6a"
        >
          <button
            aria-label="Add new Sort By"
            className="css-x1vujn"
            onClick={[Function]}
            tabIndex={0}
          >
            <div
              className="css-wf08df-Icon"
            >
              <svg
                data-testid="plus-circle"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              />
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-1f5gueu"
  >
    <label
      className="css-1e5y6i9"
    >
      Limit
      <div
        className="css-wf08df-Icon"
        tabIndex={0}
      >
        <svg
          data-testid="info-circle"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        />
      </div>
    </label>
    <div
      className="css-1rfunm5"
    >
      <div
        className="css-ax7m4m-input-wrapper"
        data-testid="input-wrapper"
      >
        <div
          className="css-1w5c5dq-input-inputWrapper"
        >
          <input
            className="css-1n6y0bv-input-input"
            max={50000}
            min={0}
            onChange={[Function]}
            style={
              {
                "paddingLeft": undefined,
                "paddingRight": undefined,
              }
            }
            type="number"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-9doeo5-InlineFieldRow"
  >
    <div
      className="css-1f5gueu"
    >
      <div
        className="css-1rfunm5"
      >
        <button
          className="css-1oe1lwi-button"
          onClick={[Function]}
          type="reset"
        >
          <span
            className="css-1mhnkuh"
          >
            Clear
          </span>
        </button>
      </div>
    </div>
    <div
      className="css-1f5gueu"
    >
      <div
        className="css-1rfunm5"
      >
        <button
          className="css-td06pi-button"
          onClick={[Function]}
          type="submit"
        >
          <span
            className="css-1mhnkuh"
          >
            Apply
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
